### PR TITLE
BOOL -> bool in compiler.h #48317

### PR DESCRIPTION
**Pull request by JeremyEspresso**
This changes `BOOL` to `bool` for all methods that still had `BOOL` as a return type in compiler.h
See #48317

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
cc @dotnet/jit-contrib 

That looks like it was fairly straightforward.

File: src/coreclr/jit/compiler.h [hunk 1]
```diff
@@ -3543,8 +3543,8 @@ class Compiler
 
     bool lvaIsParameter(unsigned varNum);
     bool lvaIsRegArgument(unsigned varNum);
-    BOOL lvaIsOriginalThisArg(unsigned varNum); // Is this varNum the original this argument?
-    BOOL lvaIsOriginalThisReadOnly();           // return TRUE if there is no place in the code
+    bool lvaIsOriginalThisArg(unsigned varNum); // Is this varNum the original this argument?
+    bool lvaIsOriginalThisReadOnly();           // return TRUE if there is no place in the code
```
Comment (BruceForstall) [inline src/coreclr/jit/compiler.h:3547]: nit: Change `TRUE` (Windows BOOL value) to `true`

File: src/coreclr/jit/importer.cpp [hunk 2]
```diff
@@ -5237,7 +5237,7 @@ GenTree* Compiler::impArrayAccessIntrinsic(
     }
 }
 
-BOOL Compiler::verMergeEntryStates(BasicBlock* block, bool* changed)
+bool Compiler::verMergeEntryStates(BasicBlock* block, bool* changed)
```
Comment (BruceForstall) [inline src/coreclr/jit/importer.cpp:5240]: All the `return` statements here need to replace `FALSE` with `false` and `TRUE` with `true`.

The same is the case with other of these functions where the return value was changed.

**Review by BruceForstall — CHANGES_REQUESTED** [review]
I pointed out a few things.

I think the change needs to be more all-in-one comprehensive, not just changing return types. It also needs to change the use of `BOOL` types in functions, and removing the use of `FALSE`/`TRUE`.

File: src/coreclr/jit/typeinfo.cpp [hunk 3]
```diff
@@ -18,7 +18,7 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
 #include "_typeinfo.h"
 
-BOOL Compiler::tiCompatibleWith(const typeInfo& child, const typeInfo& parent, bool normalisedForStack) const
+bool Compiler::tiCompatibleWith(const typeInfo& child, const typeInfo& parent, bool normalisedForStack) const
```
Comment (BruceForstall) [inline src/coreclr/jit/typeinfo.cpp:21]: below: change `BOOL compatible` => `bool compatible`

File: src/coreclr/jit/typeinfo.cpp [hunk 4]
```diff
@@ -48,12 +48,12 @@ BOOL Compiler::tiCompatibleWith(const typeInfo& child, const typeInfo& parent, b
...
 }
 
-BOOL Compiler::tiMergeCompatibleWith(const typeInfo& child, const typeInfo& parent, bool normalisedForStack) const
+bool Compiler::tiMergeCompatibleWith(const typeInfo& child, const typeInfo& parent, bool normalisedForStack) const
 {
     return typeInfo::tiMergeCompatibleWith(info.compCompHnd, child, parent, normalisedForStack);
 }
 
-BOOL Compiler::tiMergeToCommonParent(typeInfo* pDest, const typeInfo* pSrc, bool* changed) const
+bool Compiler::tiMergeToCommonParent(typeInfo* pDest, const typeInfo* pSrc, bool* changed) const
```
Comment (BruceForstall) [inline src/coreclr/jit/typeinfo.cpp:56]: below: change `BOOL mergeable` => `bool mergeable`

**Review by BruceForstall — CHANGES_REQUESTED** [review]
Thanks for the updates. Two more requests.

**Review by BruceForstall — APPROVED** [review]
LGTM. Thanks for the contribution.

**Pull request merged**